src/ThemeProvider/ThemeProvider.tsx

```diff
@@ -242,50 +231,41 @@ export default function ThemeProvider({
   children,
   theme: controlledTheme
 }: ThemeProviderProps) {
-  // theme state
-  const [theme, setTheme] = React.useState("light");
+  // wrap mui theme provider and children in theme context
+  return (
+    <MuiThemeProvider theme={mainThemeWithColorSchemes}>
+      <ChildWrapper theme={controlledTheme}>{children}</ChildWrapper>
+    </MuiThemeProvider>
+  );
+}
 
-  // effect to set the theme on mount
+/**
+ * Child wrapper to handle controlled theme changes
+ */
+function ChildWrapper({
+  children,
+  theme: controlledTheme
+}: ThemeProviderProps) {
+  // use hook from MUI to get and set the theme mode
+  const { mode, setMode } = useColorScheme();
+
+  // update the theme mode to "light" if current mode is "system"
   useEffect(() => {
-    // theme preference is decided in the following order:
-    // 1. controlled theme prop
-    // 2. local storage
-    // 3. default to light
-    const storedThemeMode = localStorage.getItem("theme");
-    if (controlledTheme !== undefined) {
-      setTheme(controlledTheme);
-    } else if (storedThemeMode !== null) {
-      setTheme(storedThemeMode);
-    } else {
-      setTheme("light");
+    if (mode === "system") {
```

Comment:
Dafault mode in MUI v6 is "system", we only have switch for dark and light and we dont relay on system mode to set app mode. To achieve the current hahaviour on QA setting mode manually to "light"